flac.go

```diff
@@ -159,16 +186,15 @@ func (stream *Stream) skipID3v2() error {
 // Stream.ParseNext to parse the entire next frame including audio samples.
 func Parse(r io.Reader) (stream *Stream, err error) {
```

Comment:
> The names New and Parse don't accurately reflect what they do. At least the functionality is well explained in the documentation. What did we think here @mewmew haha?

Haha, this is part of the great API upheaval issue in v2.0 (issue #33).

I think especially your comment (https://github.com/mewkiz/flac/pull/43/files#r559845344) about the change on contract for `New` is important to take into consideration when thinking about v2.0.

We can also try to form a consensus on what most users would find useful, what are the most likely use case and make those default. So long as we support API for doing less processing if the user really wants to avoid unnecessary parsing. That was the original idea of `New`, to support less processing. The idea from this part of the API was shamelessly stolen from how `debug/pe`, `debug/elf` does its parsing of binary executables. [pe.NewFile](https://golang.org/pkg/debug/pe/#NewFile) does not do parsing, but simply prepares a "File handle" for access. Whereas [pe.Open](https://golang.org/pkg/debug/pe/#Open) also prepares it for use as a PE file. In other words, pe.NewFile does [not do parsing](https://golang.org/src/debug/pe/file.go?s=935:972#L25), while pe.Open [does parsing](https://golang.org/src/debug/pe/file.go?s=1594:1636#L54).